subprojects/files/src/main/java/org/gradle/internal/file/impl/DefaultFileMetadata.java

```diff
@@ -20,20 +20,28 @@
 import org.gradle.internal.file.FileType;
 
 public class DefaultFileMetadata implements FileMetadataSnapshot {
-    private static final FileMetadataSnapshot DIR = new DefaultFileMetadata(FileType.Directory, 0, 0);
-    private static final FileMetadataSnapshot MISSING = new DefaultFileMetadata(FileType.Missing, 0, 0);
+    private static final FileMetadataSnapshot DIR = new DefaultFileMetadata(FileType.Directory, 0, 0, false);
+    private static final FileMetadataSnapshot SYMLINKED_DIR = new DefaultFileMetadata(FileType.Directory, 0, 0, true);
+    private static final FileMetadataSnapshot MISSING = new DefaultFileMetadata(FileType.Missing, 0, 0, false);
+    private static final FileMetadataSnapshot BROKEN_SYMLINK = new DefaultFileMetadata(FileType.Missing, 0, 0, true);
     private final FileType type;
     private final long lastModified;
     private final long length;
+    private final boolean isSymlink;
 
-    public DefaultFileMetadata(FileType type, long lastModified, long length) {
+    public DefaultFileMetadata(FileType type, long lastModified, long length, boolean isSymlink) {
         this.type = type;
         this.lastModified = lastModified;
         this.length = length;
+        this.isSymlink = isSymlink;
```

Comment:
Does this represent the thing itself being a symlink, or whether it was accessed via a symlink?

Reply:
It was accessed via a symlink. See the improvements in https://github.com/gradle/gradle/issues/13053.